draft-edns-otel-trace-ids.md

```diff
@@ -31,70 +31,167 @@ author:
     /                               /
     +---------------+---------------+
 
 ~~~
 
 Version (1 octet) has the value 0 for this specification.
-Reserved (1 octet) MUST also be set to 0.
-The Trace ID is 16 octets long and is mandatory (or not? if empty, receiver selects one, so that empty is just a "please trace" signal. Although then the sender could also just generate a trace ID).
-The optional Span ID is 8 octets long.
+Trace ID Type (1 octet) describes how the Trace ID data should be interpreted.
+Trace ID Data has a variable length, depending on the Trace ID Type.
+
+# Presentation Format
+
+Even though EDNS options will never appear in DNS zone files, its value could appear in logging or analysis of packet captures.
+The presentation format for TRACEID is as follows:
+
+~~~ ascii-art
+TRACEID=[mnemonic]:[data]
+~~~
+
+Where the `mnemonic` is the mnemonic defined for the ID Type and `data` is the ID Data represented as Base16.
+When an unknown Type is encountered, the `mnemonic` MUST be presented as `TYPEN`, where `N` is the decimal representation of the Trace ID Type without leading zeroes.
+
+# Trace ID Types
+
+This specification defines several values for Trace ID Type.
+
+
+| Type name     | Mnemonic      | Trace ID Type |
+| ------------- | ------------- | ------------- |
+| OpenTelemetry | OT            | 0             |
+| Private use   | PRIVATENNN (where NNN is the decimal representation of the Type) | 247 - 254     |
+| RESERVED      | RESERVED           | 255           |
+
+
+## OpenTelemetry (0)
+
+OpenTelemetry{{OT.WEBSITE}} is an open standard for telemetry data like metrics, logs and traces.
+It is maintained by the Cloud Native Computing Foundation (CNCF){{CNCF.WEBSITE}}.
+
+For OpenTelemetry traces, the TraceID Data field MUST contain a 16 octet Trace ID and MAY have an 8 octet Span ID following it.
+This makes the Trace ID data field either 16 or 24 octets long.
+
+## Private use (247 - 254)
+
+These Trace ID Types can be used for private tracing identifiers.
+
+## RESERVED (255)
+
+This Trace ID Type value is reserved for potential future expansion and MUST NOT be used.
+
+# Processing of TRACEID
+
+TRACEID SHOULD only be used after mutual agreement between the upstream and downstream server operators.
+
+Performing tracing SHOULD NOT impact DNS query processing.
+Hence, nameservers receiving a malformed TRACEID option or a TRACEID option with an unknown or unsupported Type ID SHOULD ignore this option an continue processing the query.
+It is RECOMMENDED to inform the operator of the nameserver, for example using logging, about malformed or unknown TRACEID options.
+
+Tracing information is collected outside of the DNS transaction and is independent of the DNS query processing.
+The inclusion of a TRACEID option in a query must be seen as a one-way signal from the requestor that tracing should be performed.
+Responders MUST NOT include a TRACEID option in responses, even to queries that contained a TRACEID option.
```

Comment:
I was thinking about the potential use case: TraceID/SpanID is generated by server and sent to client in a response.
The user of the client software then immediately knows which TraceID to look for in a collector.
This is no longer allowed with this MUST NOT and forces the client to generate the ID(s).

Reply:
we could perhaps define separate behavior when the Trace ID Data is empty. I'll type that up.